### PR TITLE
Store: Add a few objectives for Store's data touched/fetched amount and sizes

### PR DESCRIPTION
**Pull request by douglascamata**
Signed-off-by: Douglas Camata <159076+douglascamata@users.noreply.github.com>

* [x] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

* Add a few objectives for the summaries reporting the quantity and size of items fetched per data type from object storage & cache. They are: 50, 95, and 99 quantiles. 

The intent is to provide some metrics that will help inform users when setting values for the flags `store.grpc.series-sample-limit` and `store.grpc.touched-series-limit`.

## Verification

- Ran it locally and confirmed the metrics are there and producing the configured quantiles.

### CODE REVIEW

File: pkg/store/bucket.go [hunk 1]
```diff
@@ -168,21 +168,25 @@ func newBucketStoreMetrics(reg prometheus.Registerer) *bucketStoreMetrics {
 	})
 
 	m.seriesDataTouched = promauto.With(reg).NewSummaryVec(prometheus.SummaryOpts{
-		Name: "thanos_bucket_store_series_data_touched",
-		Help: "How many items of a data type in a block were touched for a single series request.",
+		Name:       "thanos_bucket_store_series_data_touched",
+		Help:       "How many items of a data type in a block were touched for a single series request.",
```
Comment (moadz) [inline pkg/store/bucket.go:172]:
```suggestion
		Help:       "Number of series touched to fulfil a single Store API series request.",
```

File: pkg/store/bucket.go [hunk 2]
```diff
@@ -168,21 +168,25 @@ func newBucketStoreMetrics(reg prometheus.Registerer) *bucketStoreMetrics {
...
+		Objectives: map[float64]float64{0.50: 0.1, 0.95: 0.1, 0.99: 0.001},
 	}, []string{"data_type"})
 	m.seriesDataFetched = promauto.With(reg).NewSummaryVec(prometheus.SummaryOpts{
-		Name: "thanos_bucket_store_series_data_fetched",
-		Help: "How many items of a data type in a block were fetched for a single series request.",
+		Name:       "thanos_bucket_store_series_data_fetched",
+		Help:       "How many items of a data type in a block were fetched for a single series request.",
```
Comment (moadz) [inline pkg/store/bucket.go:177]:
```suggestion
		Help:       "Number of series retrieved to fulfil a single Store API series request.",
```

File: pkg/store/bucket.go [hunk 3]
```diff
@@ -168,21 +168,25 @@ func newBucketStoreMetrics(reg prometheus.Registerer) *bucketStoreMetrics {
...
-		Name: "thanos_bucket_store_series_data_touched",
-		Help: "How many items of a data type in a block were touched for a single series request.",
+		Name:       "thanos_bucket_store_series_data_touched",
+		Help:       "How many items of a data type in a block were touched for a single series request.",
+		Objectives: map[float64]float64{0.50: 0.1, 0.95: 0.1, 0.99: 0.001},
 	}, []string{"data_type"})
 	m.seriesDataFetched = promauto.With(reg).NewSummaryVec(prometheus.SummaryOpts{
-		Name: "thanos_bucket_store_series_data_fetched",
-		Help: "How many items of a data type in a block were fetched for a single series request.",
+		Name:       "thanos_bucket_store_series_data_fetched",
```
Comment (moadz) [inline pkg/store/bucket.go:176]: Data is redundant. 

```suggestion
		Name:       "thanos_bucket_store_series_fetched",
```

File: pkg/store/bucket.go [hunk 4]
```diff
@@ -168,21 +168,25 @@ func newBucketStoreMetrics(reg prometheus.Registerer) *bucketStoreMetrics {
 	})
 
 	m.seriesDataTouched = promauto.With(reg).NewSummaryVec(prometheus.SummaryOpts{
-		Name: "thanos_bucket_store_series_data_touched",
-		Help: "How many items of a data type in a block were touched for a single series request.",
+		Name:       "thanos_bucket_store_series_data_touched",
```
Comment (moadz) [inline pkg/store/bucket.go:171]: Data is redundant. 

```suggestion
		Name:       "thanos_bucket_store_series_touched",
```

File: pkg/store/bucket.go [hunk 5]
```diff
@@ -168,21 +168,25 @@ func newBucketStoreMetrics(reg prometheus.Registerer) *bucketStoreMetrics {
...
-		Help: "How many items of a data type in a block were fetched for a single series request.",
+		Name:       "thanos_bucket_store_series_data_fetched",
+		Help:       "How many items of a data type in a block were fetched for a single series request.",
+		Objectives: map[float64]float64{0.50: 0.1, 0.95: 0.1, 0.99: 0.001},
 	}, []string{"data_type"})
 
 	m.seriesDataSizeTouched = promauto.With(reg).NewSummaryVec(prometheus.SummaryOpts{
-		Name: "thanos_bucket_store_series_data_size_touched_bytes",
-		Help: "Size of all items of a data type in a block were touched for a single series request.",
+		Name:       "thanos_bucket_store_series_data_size_touched_bytes",
```
Comment (moadz) [inline pkg/store/bucket.go:182]:
```suggestion
		Name:       "thanos_bucket_store_series_touched_bytes",
```

File: pkg/store/bucket.go [hunk 6]
```diff
@@ -168,21 +168,25 @@ func newBucketStoreMetrics(reg prometheus.Registerer) *bucketStoreMetrics {
...
+		Help:       "Size of all items of a data type in a block were touched for a single series request.",
```
Comment (moadz) [inline pkg/store/bucket.go:183]:
```suggestion
		Help:       "Total size of series touched to fulfil a single Store API series request in Bytes.",
```

File: pkg/store/bucket.go [hunk 7]
```diff
@@ -168,21 +168,25 @@ func newBucketStoreMetrics(reg prometheus.Registerer) *bucketStoreMetrics {
...
+		Objectives: map[float64]float64{0.50: 0.1, 0.95: 0.1, 0.99: 0.001},
 	}, []string{"data_type"})
 	m.seriesDataSizeFetched = promauto.With(reg).NewSummaryVec(prometheus.SummaryOpts{
-		Name: "thanos_bucket_store_series_data_size_fetched_bytes",
-		Help: "Size of all items of a data type in a block were fetched for a single series request.",
+		Name:       "thanos_bucket_store_series_data_size_fetched_bytes",
```
Comment (moadz) [inline pkg/store/bucket.go:187]:
```suggestion
		Name:       "thanos_bucket_store_series_fetched_bytes",
```

File: pkg/store/bucket.go [hunk 8]
```diff
@@ -168,21 +168,25 @@ func newBucketStoreMetrics(reg prometheus.Registerer) *bucketStoreMetrics {
...
+		Help:       "Size of all items of a data type in a block were fetched for a single series request.",
```
Comment (moadz) [inline pkg/store/bucket.go:188]:
```suggestion
		Help:       "Total size of series fetched to fulfil a single Store API series request in Bytes.",
```

File: pkg/store/bucket.go [hunk 9]
```diff
@@ -200,8 +204,9 @@ func newBucketStoreMetrics(reg prometheus.Registerer) *bucketStoreMetrics {
 		Buckets: []float64{0.001, 0.01, 0.1, 0.3, 0.6, 1, 3, 6, 9, 20, 30, 60, 90, 120},
 	})
 	m.resultSeriesCount = promauto.With(reg).NewSummary(prometheus.SummaryOpts{
-		Name: "thanos_bucket_store_series_result_series",
-		Help: "Number of series observed in the final result of a query.",
+		Name:       "thanos_bucket_store_series_result_series",
```
Comment (moadz) [inline pkg/store/bucket.go:207]: This is redundant as the store-api metric will illustrate this data for a single query across all stores, and the store-specific metric you added for `thanos_bucket_store_series_touched` covers the to total fetched for a single storeAPI request.

**Review by moadz — CHANGES_REQUESTED** [review]

**Comment by douglascamata** [comment]
@moadz I would love to simplify and make the metric names shorter, but I don't think it's a good idea to change their names after they have been out since many releases. The reason is that people might be already relying on them for dashboards (i.e. [kube-thanos](https://github.com/thanos-io/kube-thanos) uses them) and alerts.

**Review by moadz — APPROVED** [review]
@douglascamata you're absolutely right. I misread the diff and thought you were adding them as new metrics :) Bucket changes LGTM

**Review by PhilipGough — APPROVED** [review]
lgtm

**Review by bwplotka — APPROVED** [review]
Nice, thanks!

File: CHANGELOG.md [hunk 10]
```diff
@@ -34,6 +34,7 @@ We use *breaking :warning:* to mark changes that are not backward compatible (re
 ### Removed
 
 - [#5824](https://github.com/thanos-io/thanos/pull/5824) Mixin: Remove noisy `ThanosReceiveTrafficBelowThreshold` alert.
+- [#5819](https://github.com/thanos-io/thanos/pull/5819) Store: Add a few objectives for Store's data touched/fetched amount and sizes. They are: 50, 95, and 99 quantiles.
```
Comment (matej-g) [inline CHANGELOG.md:37]: I guess this was accidentally put into removed section instead of added?

Comment (douglascamata) [reply CHANGELOG.md:37]: Oh, my bad, @matej-g. Fixed in https://github.com/thanos-io/thanos/pull/5919.

**Pull request merged**